src/amrex_mini_app/src/field/field.cpp

```diff
@@ -0,0 +1,214 @@
+
+    const amrex::Box box(lower_index, upper_index, index_type);
+
+    amrex::BoxArray box_array(box);
+
+    // Break up boxarray "cell_box_array" into chunks no larger than "max_chunk_size" along a direction
+    const int max_chunk_size = 32; // Hardcoded for now, but could be a parameter for the user to set via the constructor arguments.
+    box_array.maxSize(max_chunk_size);
+
+    const amrex::DistributionMapping distribution_mapping(box_array);
+
+    multifab = std::make_shared<amrex::MultiFab>(box_array, distribution_mapping, n_component, n_ghost);
+
+}
+
+bool Field::IsCellCentered() const noexcept{
+    return (field_grid_stagger == FieldGridStagger::CellCentered);
+}
+
+bool Field::IsIFaceCentered() const noexcept {
+    return (field_grid_stagger == FieldGridStagger::IFace);
+}
+
+bool Field::IsJFaceCentered() const noexcept{
+    return (field_grid_stagger == FieldGridStagger::JFace);
+}
+
+bool Field::IsKFaceCentered() const noexcept {
+    return (field_grid_stagger == FieldGridStagger::KFace);
+}
+
+bool Field::IsNodal() const noexcept {
+    return (field_grid_stagger == FieldGridStagger::Nodal);
+}
+
+// This is where the coupling between the Field and Grid classes happens
+Grid::Point Field::GetGridPoint(int i, int j, int k) const {
+    switch(field_grid_stagger) {
+        case FieldGridStagger::CellCentered:
+            return grid->CellCenter(i, j, k);
+        case FieldGridStagger::IFace:
+            return grid->IFace(i, j, k);
+        case FieldGridStagger::JFace:
+            return grid->JFace(i, j, k);
+        case FieldGridStagger::KFace:
+            return grid->KFace(i, j, k);
+        case FieldGridStagger::Nodal:
+            return grid->Node(i, j, k);
+        default:
+            throw std::invalid_argument("Field::GetGridPoint: Invalid FieldGridStagger specified.");
+    }
+}
+
+// Write the field data to an HDF5 file. This will overwrite the file if it already exists.
+void Field::WriteHDF5(const std::string& filename) const {
+    const hid_t file_id = H5Fcreate(filename.c_str(), H5F_ACC_TRUNC, H5P_DEFAULT, H5P_DEFAULT);
+    if (file_id < 0) {
+        throw std::runtime_error("Field::WriteHDF5: Failed to create HDF5 file: " + filename);
+    }
+    WriteHDF5(file_id);
+    H5Fclose(file_id);
+}
+
+// Write the field data to an already open HDF5 file that you already have open.
+void Field::WriteHDF5(const hid_t file_id) const {
+
+    if (file_id < 0) {
+        throw std::runtime_error("Field::WriteHDF5: Invalid HDF5 file_id passed to WriteHDF5.");
+    }
+
+    // Copy the MultiFab to a single rank
+    int dest_rank = 0; // We are copying to rank 0
+    const std::shared_ptr<amrex::MultiFab> mf = CopyMultiFabToSingleRank(multifab, dest_rank);
+
+    if (amrex::ParallelDescriptor::MyProc() == dest_rank) {
+
+        AMREX_ASSERT(mf->boxArray().size() == 1);
+        amrex::Box box = mf->boxArray()[0]; // We are assuming that there is only one box in the MultiFabs box array.
```

Comment:
We are already baking in the assumption that there is only a single versus the full flexibility of mulitple boxes.  Won't we need a more flexible implmentation to make sure we don't break AMR capability.  If we don't want to fix this issue now, maybe we add in something that generates an error if it has multiple boxes.